util/network/src/client_version.rs

```diff
@@ -0,0 +1,473 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+// Parity client string prefix
+const LEGACY_CLIENT_ID_PREFIX: &str = "Parity";
+const PARITY_CLIENT_ID_PREFIX: &str = "Parity-Ethereum";
+
+// Parity versions starting from this will accept block bodies requests
+// of 256 bodies
+const PARITY_CLIENT_LARGE_REQUESTS_VERSION: &str = "2.3.0";
+
+// Parity versions starting from this will accept service-transactions
+const SERVICE_TRANSACTIONS_VERSION: &str = "1.6.0";
```

Comment:
That means getting rid of the `accepts_service_transaction` check as well, and in `SyncPropagator::propagate_new_transactions` simply pass a tautological filter to `select_peers_for_transactions`. That would leave no use for the filter parameter and it could simply be removed, right?

Reply:
> That means getting rid of the accepts_service_transaction check as well,

Yes (although see below)

> in SyncPropagator::propagate_new_transactions simply pass a tautological filter to select_peers_for_transactions

We should still send only to Parity Ethereum peers, so might be easier to have `accepts_service_transaction` return `true` only for Parity Ethereum (no matter what version), but not Geth/other impls